File: core/core-domain/src/main/java/com/coffee/gu/coupon/CouponTargetRepository.java
package com.coffee.gu.coupon;

import com.coffee.gu.enums.CouponTargetType;

import java.util.List;

public interface CouponTargetRepository {
    List<CouponTarget> findAllByTargetTypeAndTargetIdIn(CouponTargetType targetType, List<Long> targetIds);
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/coupon/CouponTargetRepository.kt
package com.coffee.gu.coupon

import com.coffee.gu.enums.CouponTargetType

interface CouponTargetRepository {
    fun findAllByTargetTypeAndTargetIdIn(targetType: CouponTargetType, targetIds: List<Long>): List<CouponTarget>
}
